src/main/kotlin/example/ColorMode.kt

```diff
@@ -0,0 +1,45 @@
+package example
+
+import kotlinext.js.jso
+import mui.material.styles.*
+import react.*
+
+enum class PaletteMode(val paletteOptions: PaletteOptions) {
+    Light(
+        jso {
+            mode = "light"
+        }
+    ),
+    Dark(
+        jso {
+            mode = "dark"
+        }
+    )
+}
+
+typealias ColorModeState = StateInstance<PaletteMode>
+
+val ColorModeContext = createContext<ColorModeState>()
+
+val ColorModeModule = FC<PropsWithChildren> { props ->
```

Comment:
Please create single `ThemeModule` which encapsulates MUI `ThemeProvider` and your `ColorModeContext` (also plese rename it to `ThemeContext`).

Reply:
You mean something like it? Or an another way to do the double encapsulation?

```kotlin
val ThemeModule = FC<PropsWithChildren> { props ->
    val theme = useState(Themes.Light)

    ThemeContext.Provider(theme) {
        ThemeProvider {
            this.theme = theme
            props.children()
        }
    }
}
```

Reply:
Exactly this